src/test/scala/com/codacy/parsers/CloverParserTest.scala

```diff
@@ -3,124 +3,189 @@ package com.codacy.parsers
-        val reader = CloverParser.parse(new File("."), new File(nonExistentReportPath))
+      "the report is not in the Clover format" in {
+        // Arrange
+        val reportNotInCloverFormat = "src/test/resources/test_cobertura.xml"
+
+        // Act
+        val parseResult = CloverParser.parse(new File("."), new File(reportNotInCloverFormat))
 
-        reader shouldBe 'left
+        // Assert
+        parseResult shouldBe 'left
       }
 
-      "file does not follow the Clover schema" in {
-        val reader = CloverParser.parse(new File("."), new File(invalidCloverReportPath))
-        reader shouldBe 'left
+      "the report is missing the statements attribute in the file metrics element" in {
+        // Arrange
+        val invalidCloverReportPath = "src/test/resources/test_invalid_clover.xml"
+
+        // Act
+        val parseResult = CloverParser.parse(new File("."), new File(invalidCloverReportPath))
+
+        // Assert
+        parseResult shouldBe 'left
       }
+
     }
 
-    "identify the report as valid" when {
-      "report has packages" in {
-        val reader = CloverParser.parse(new File("."), new File(cloverReportPath))
-        reader shouldBe 'right
+    val cloverReportPath = "src/test/resources/test_clover.xml"
+    val cloverWithoutPackagesFilePath = "src/test/resources/test_clover_without_packages.xml"
+
+    "succeed to parse a valid report" when {
+
+      "the report has packages" in {
+        // Act
+        val parseResult = CloverParser.parse(new File("."), new File(cloverReportPath))
+
+        // Assert
+        parseResult shouldBe 'right
       }
 
-      "report does not have packages" in {
-        val readerWithoutPackages = CloverParser
+      "the report does not have packages" in {
+        // Act
+        val parseResult = CloverParser
           .parse(new File("/home/codacy-php/"), new File(cloverWithoutPackagesFilePath))
 
-        readerWithoutPackages shouldBe 'right
+        // Assert
+        parseResult shouldBe 'right
       }
+
     }
 
     "parse correct file paths" when {
+
       "reports contain both name and path attributes in file elements" in {
+        // Arrange
         val cloverWithPaths = new File("src/test/resources/test_clover_with_paths.xml")
 
-        val readerWithoutPackages =
-          CloverParser.parse(new File("/Users/username/workspace/repository"), cloverWithPaths)
-
-        readerWithoutPackages shouldBe 'right
-        readerWithoutPackages.right.value.fileReports.head.filename shouldEqual "src/app/file.js"
+        // Act
+        val parsedReportFilePaths =
+          CloverParser
+            .parse(new File("/Users/username/workspace/repository"), cloverWithPaths)
+            .right
+            .value
+            .fileReports
+            .map(_.filename)
+
+        // Assert
+        parsedReportFilePaths should contain("src/app/file.js")
       }
+
     }
 
     "return the same report with or without packages" in {
```

Comment:
Doesn't this test replace the ones above about packages?

Reply:
Not really, the others test that you can successfully parse with and without packages, this one tests specifically that they have the same values.